### PR TITLE
connected context with db

### PR DESCRIPTION
**Pull request by Jatinchhabra21**
Ticket: [BOOKA-15](https://bookkeeper-manager.atlassian.net/browse/BOOKA-15)

[BOOKA-15]: https://bookkeeper-manager.atlassian.net/browse/BOOKA-15?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BookkeeperAPI/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    public class UserController : ControllerBase
+    {
+        // TODO: Move business logic to service layer and Database logic to Repository layer
+        private BookkeeperContext _context;
+        public UserController(BookkeeperContext context)
+        {
+            _context = context;
+        }
+
+        [HttpGet("/api/me/account")]
```
Comment (hritiksingh13) [inline BookkeeperAPI/Controllers/UserController.cs:25]: Include user Id in the URL

File: BookkeeperAPI/Entity/Currency.cs
Comment (hritiksingh13) [inline BookkeeperAPI/Entity/Currency.cs:1]: Create a separate folder and file which will contain all constants and enums.

File: BookkeeperAPI/Entity/ExpenseCategory.cs
Comment (hritiksingh13) [inline BookkeeperAPI/Entity/ExpenseCategory.cs:1]: Create a separate folder and file which will contain all constants and enums.

File: BookkeeperAPI/Entity/Theme.cs
Comment (hritiksingh13) [inline BookkeeperAPI/Entity/Theme.cs:1]: Same as above comment for this one as well.

File: BookkeeperAPI/Program.cs [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+builder.Services.AddControllers()
+    .AddNewtonsoftJson(options =>
+    {
+        options.SerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
+    }); ;
+
+// Add DB context to connect to database
+builder.Services.AddDbContext<BookkeeperContext>(
+    optionsBuilder => optionsBuilder
+    .UseNpgsql(configuration["ConnectionStrings:BookkeeperDB"])
```
Comment (hritiksingh13) [inline BookkeeperAPI/Program.cs:22]: value for `ConnectionStrings:BookkeeperDB` is missing from appsettings.json. We should store in a keyvault(If possible).

**Review by hritiksingh13 — CHANGES_REQUESTED** [review]
Added some comments, after this we are good to merge !!!

File: BookkeeperAPI/Entity/UserCredential.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        [Column("last_updated")]
+        public DateTime LastUpdated { get; set; } 
+
+        [Column("created_time")]
+        public DateTime CreatedTime { get; set; }
+
+
+
+        [Column("user_id")]
+        public Guid UserId { get; set; }
```
Comment (nikhilk839) [inline BookkeeperAPI/Entity/UserCredential.cs:39]: Please check if this is a foreign key or not, if no then in that case I think the Delete cascade will not work

Comment (Jatinchhabra21) [reply BookkeeperAPI/Entity/UserCredential.cs:39]: It is a foreign key to users table

File: BookkeeperAPI/Program.cs
Comment (nikhilk839) [inline BookkeeperAPI/Program.cs:1]: I will suggest including an exception handler middleware that will able to log any unhandled exception.

File: BookkeeperAPI/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                Email = request.Email,
+                LastUpdated = DateTime.UtcNow,
+                CreatedTime = DateTime.UtcNow,
+            };
+            _context.Credentials.Add(user.Credential);
+            _context.SaveChanges();
+            return StatusCode(201, u.Entity);
+        }
+
+        [HttpPut("/api/users/{userId}/preference")]
```
Comment (nikhilk839) [inline BookkeeperAPI/Controllers/UserController.cs]: Use Patch instead of PUT

File: BookkeeperAPI/Data/BookkeeperContext.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        public DbSet<User> Users { get; set; }
+
+        public DbSet<UserCredential> Credentials { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<User>()
+                .HasMany(e => e.Expenses)
+                .WithOne(u => u.User)
+                .HasForeignKey(e => e.UserId)
```
Comment (nikhilk839) [inline BookkeeperAPI/Data/BookkeeperContext.cs:24]: Please check relation with expense table

Comment (Jatinchhabra21) [reply BookkeeperAPI/Data/BookkeeperContext.cs:24]: connected offline

File: BookkeeperAPI/Program.cs (hunk 2 above)
Comment (Jatinchhabra21) [reply BookkeeperAPI/Program.cs:22]: currently, this is being stored in appsettings.Development.json, will create keyvault soon and integrate it in current build

Comment (Jatinchhabra21) [reply BookkeeperAPI/Program.cs:22]: To track keyvault progress, created a task, [BOOKA-39](https://bookkeeper-manager.atlassian.net/browse/BOOKA-39)

File: BookkeeperAPI/Controllers/UserController.cs (hunk 4 above)
Comment (Jatinchhabra21) [reply BookkeeperAPI/Controllers/UserController.cs]: updated

File: BookkeeperAPI/Program.cs
Comment (Jatinchhabra21) [reply BookkeeperAPI/Program.cs:1]: created basic custom middleware for handling exceptions

**Pull request merged**